wasm/main.go

```diff
@@ -0,0 +1,62 @@
+//go:build js && wasm
+
+package main
+
+import (
+	"crypto/ed25519"
+	"encoding/base64"
+	"encoding/json"
+	"syscall/js"
+
+	"github.com/TBD54566975/ssi-sdk/crypto"
+	"github.com/TBD54566975/ssi-sdk/did"
+)
+
+/*
+ * This is the glue to bind the functions into javascript so they can be called
+ */
+func main() {
+	done := make(chan struct{})
+
+	// Bind the functions to javascript
+	js.Global().Set("sayHello", js.FuncOf(sayHello))
+	js.Global().Set("generateKey", js.FuncOf(generateKey))
+	js.Global().Set("makeDid", js.FuncOf(makeDid))
+
+	<-done
+}
+
+// 1. Simplest function - note we wrap things with js.ValueOf (if a primitive you don't technically need to)
+func sayHello(_ js.Value, args []js.Value) interface{} {
+	return js.ValueOf("Hello from golang via wasm!")
+}
+
+// 2. Calling a ssi-sdk function directly - but returning a plain old string
+func generateKey(_ js.Value, args []js.Value) interface{} {
+
+	keyType := args[0].String()
+	for _, k := range crypto.GetSupportedKeyTypes() {
+		if string(k) == keyType {
+			publicKey, _, _ := crypto.GenerateKeyByKeyType(k)
+			pubKeyBytes, _ := crypto.PubKeyToBytes(publicKey)
+			return js.ValueOf(base64.StdEncoding.EncodeToString(pubKeyBytes))
+		}
+	}
+
+	return js.ValueOf("Unknown key type")
```

Comment:
Consider using `IsSupportedKeyType` and returning early to reduce nesting. 
```suggestion
        kt := crypto.KeyType(keyType)
	if !crypto.IsSupportedKeyType(kt) {
	  return js.ValueOf("Unknown key type")
	}
	publicKey, _, _ := crypto.GenerateKeyByKeyType(kt)
	pubKeyBytes, _ := crypto.PubKeyToBytes(publicKey)
	return js.ValueOf(base64.StdEncoding.EncodeToString(pubKeyBytes))
```